tests/main/interfaces-contacts-service/task.yaml

```diff
@@ -7,7 +7,7 @@ systems:
     - -ubuntu-14.04-*
 
 environment:
-    XDG: $(pwd)/xdg
+    XDG: $(mktemp -d -p $(pwd) xdgXXXXXXXX)
```

Comment:
I can't see how that would fix some kind of flakiness in this test. The original $(pwd)/xdg directory doesn't exist in the tree, does it?  If so, how would this test possibly execute twice for it to make a difference that this is set to something elsewhere?

What is the error message like?

Reply:
I see, I don't understand all the details of how tests are instantiated by spread (and in particular what's possibly reused and what's not, and if `-repeat` reuses anything as opposed to normal runs), so misunderstanding on my side on that part. The error we occasionally see (address book backend fails on adding a new contact complaining that it already exists) made me conclude it's a problem of ~/xdg not getting cleaned up properly sometimes.
I'm closing it for now as I've run out of ideas for the time being.